### PR TITLE
GUI - Captain's Log

### PR DESCRIPTION
**Pull request by dalderliesten**
After a lot of hassle, finally got a simple heads-up display (HUD) to work in the form of a captain's log.  The log displays the active events in the game client and updates appropriately.  The rest of the GUI, whose work has been forced to be scrapped, will come later.

Also refactored parts of the game class.

Also added some timing items to the event class for future use.

### CODE REVIEW

**Comment by dalderliesten** [comment]
This PR addresses issues #100 and some of #107.

File: src/main/java/nl/tudelft/pixelperfect/event/Event.java [hunk 1]
```diff
@@ -104,6 +112,16 @@ public boolean isExpired(long time) {
...
 
   /**
+   * Updates the time left variable for the event.
+   * 
+   * @param currentTime
+   *          the current time of the game
+   */
+  public void updateTimeLeft(long currentTime) {
+    this.timeLeft = this.timestamp - currentTime;
+  }
```
Comment (wouterz) [inline src/main/java/nl/tudelft/pixelperfect/event/Event.java]: Why seperate updateTimeLeft and getTimeLeft, and not just do this simple calculation in the getter

Comment (dmalarev) [inline src/main/java/nl/tudelft/pixelperfect/event/Event.java]: you might want to make a quick test case for this.

**Comment by dmalarev** [comment]
Rest looks good

File: src/main/java/nl/tudelft/pixelperfect/Constants.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+/**
+ * A data class for storing game-wide items that can all be altered in a single sweep.
+ * 
+ * @author David Alderliesten
+ *
+ */
+public final class Constants {
+  // Gui-styling related constants for the in-game HUD.
+  public static final int guiWidthOffset = 150;
+  public static final int guiHeightOffset = 50;
```
Comment (wouterz) [inline src/main/java/nl/tudelft/pixelperfect/Constants.java]: I believe naming of static finals should be ALL_CAPS

File: src/main/java/nl/tudelft/pixelperfect/event/Event.java (hunk 1 above)
Comment (dalderliesten) [inline src/main/java/nl/tudelft/pixelperfect/event/Event.java]: Thanks for catching this, i'll fix it.

File: src/main/java/nl/tudelft/pixelperfect/Constants.java (hunk 2 above)
Comment (dalderliesten) [inline src/main/java/nl/tudelft/pixelperfect/Constants.java]: Indeed, but an exception is allowed to be made for a constants class, see the lack of checkstyle warnings about it.

Comment (wouterz) [inline src/main/java/nl/tudelft/pixelperfect/Constants.java]: I do get checkstyle warnings about it, copied from maven site:

 Warning	naming	ConstantName	Name 'guiWidthOffset' must match pattern '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'.	11
 Warning	naming	ConstantName	Name 'guiHeightOffset' must match pattern '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'.	12
 Warning	naming	ConstantName	Name 'noEventsLogText' must match pattern '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'.	15

Comment (dalderliesten) [inline src/main/java/nl/tudelft/pixelperfect/Constants.java]: Gotcha, fixing that too.

**Comment by wouterz** [comment]
Few checkstyle warnings, 3 naming issues and 2 methods over 30 lines. Change or suppress these

File: src/main/java/nl/tudelft/pixelperfect/gui/GameHeadsUpDisplay.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+   * 
+   * @param passedMan
+   *          the passed assetmanager.
+   * @param passedGuiNode
+   *          the passed gui node.
+   * @param passedWid
+   *          the passed screen width.
+   * @param passedHi
+   *          the passed screen height.
+   */
```
Comment (dmalarev) [inline src/main/java/nl/tudelft/pixelperfect/gui/GameHeadsUpDisplay.java]: Missing param tag for passedShip

Comment (dalderliesten) [inline src/main/java/nl/tudelft/pixelperfect/gui/GameHeadsUpDisplay.java]: Tagged, working on it.

File: src/main/java/nl/tudelft/pixelperfect/event/Event.java (hunk 1 above)
Comment (jessetilro) [inline src/main/java/nl/tudelft/pixelperfect/event/Event.java]: Don't use System.currentTimeMillis(), just accept a "now" timestamp as a parameter. Makes testing easier.

**Comment by dalderliesten** [comment]
Fixed all the issues presented, open for re-review.

File: src/main/java/nl/tudelft/pixelperfect/Game.java [hunk 4]
```diff
@@ -27,6 +27,7 @@
 
 import java.io.IOException;
 
+
```
Comment (dalderliesten) [inline src/main/java/nl/tudelft/pixelperfect/Game.java]: You're a steap eahead of me, thanks.

**Comment by wouterz** [comment]
fixed last checkstyle error, should be good now. Any final comments before merging?

**Comment by dmalarev** [comment]
I'm getting exceptions while initializing the game under this branch

**Comment by dmalarev** [comment]
Everything looks good, i approve

**Pull request merged**